src/common/assets/domain/entities/__tests__/asset.unit.test.ts

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { deserialize, ObjectSchema } from 'atomicassets';
+import { Long } from 'mongodb';
+import { Asset, AssetData } from '../asset';
+
+jest.mock('atomicassets');
+const ObjectSchemaMock = ObjectSchema as jest.MockedFunction<
+  typeof ObjectSchema
+>;
+const deserializeMock = deserialize as jest.MockedFunction<typeof deserialize>;
+
+const assetDataDto = {
+  collection_name: 'foo.collection',
+  template_id: 123456,
+  schema_name: 'foo.schema',
+  immutable_serialized_data: {
+    cardid: 12,
+    name: 'foo',
+    img: 'foo.img',
+    backimg: 'foo.backimg',
+    rarity: 'rare',
+    shine: 'stone',
+    type: 'any',
+    delay: 0,
+    difficulty: 1,
+    ease: 10,
+    luck: 100,
+  },
+};
+
+const assetDto = {
+  asset_id: Long.fromBigInt(0n),
+  owner: 'foo.owner',
+  data: assetDataDto,
+};
+
+const assetSmartContractData = {
+  collectionName: 'foo.collection',
+  schemaName: 'foo.schema',
+  templateId: 123,
+} as any;
+const templateSmartContractData = {
+  schemaName: 'foo.schema',
+  immutableSerializedData: Uint8Array.from([]),
+} as any;
+const schemaSmartContractData = {
+  format: {
+    name: 'foo.name',
+    type: 'foo.type',
+  },
+} as any;
+
+describe('AssetData unit tests', () => {
+  it('AssetData.fromDto should return AssetData object based on data provided in given dto', async () => {
+    const assetData = AssetData.fromDto(assetDataDto);
+    expect(assetData).toEqual({
+      collectionName: 'foo.collection',
+      templateId: 123456,
+      schemaName: 'foo.schema',
+      immutableSerializedData: {
```

Comment:
true.... shame ... as I write all of this I keep it in the test file because of the time and amount of work. You can check the develop branch unit tests there are schemas/mocked data in common files but now ..... uff I want to remove __mocks__ btw.